pkg/crc/machine/bundle/metadata.go

```diff
@@ -244,6 +251,34 @@ func GetBundleNameFromURI(bundleURI string) string {
 	}
 }
 
+// GetBundleInfoFromName Parses the bundle filename and returns a FilenameInfo struct
+func GetBundleInfoFromName(bundleName string) (*FilenameInfo, error) {
+	var filenameInfo FilenameInfo
+	bundleName = GetBundleNameWithoutExtension(bundleName)
+	filenameParts := strings.Split(bundleName, "_")
+
+	switch len(filenameParts) {
+	case 4:
+		filenameInfo.Preset = crcPreset.OpenShift
+		filenameInfo.Driver = filenameParts[1]
+		filenameInfo.Version = filenameParts[2]
+		filenameInfo.Arch = filenameParts[3]
+	case 5:
+		parsedPreset, err := crcPreset.ParsePresetE(filenameParts[1])
+		if err != nil {
+			return &filenameInfo, err
+		}
+		filenameInfo.Preset = parsedPreset
+		filenameInfo.Driver = filenameParts[2]
+		filenameInfo.Version = filenameParts[3]
+		filenameInfo.Arch = filenameParts[4]
+	default:
```

Comment:
hey, apologies i should've mentioned this earlier, but forgot, there's a scenario where the `filenameParts` can be 6, we have a custom bundle generation command `crc bundle generate` that creates new bundle out of existing one and it names them as `crc_microshift_vfkit_4.16.7_arm64_2345.crcbundle` for a microshift bundle and `crc_vfkit_4.16.7_arm64_2345.crcbundle` for an openshift bundle which are valid names

so if i use a custom bundle currently this fails, for an openshift bundle as:

```
% crc config set bundle ./crc_vfkit_4.16.7_arm64_2345.crcbundle
WARN Using custom bundle crc_vfkit_4.16.7_arm64_2345.crcbundle 
Successfully configured bundle to ./crc_vfkit_4.16.7_arm64_2345.crcbundle

% crc setup --log-level debug
DEBU CRC version: 2.41.0+1921db
DEBU OpenShift version: 4.16.7
DEBU MicroShift version: 4.16.7
DEBU Running 'crc setup'
DEBU Got bundle path: ./crc_vfkit_4.16.7_arm64_2345.crcbundle
DEBU Failed to parse url: parse "./crc_vfkit_4.16.7_arm64_2345.crcbundle": invalid URI for request
WARN Using custom bundle crc_vfkit_4.16.7_arm64_2345.crcbundle
Cannot parse preset 'vfkit'
```

for a microshift bundle, as:

```
% crc config set bundle ./crc_microshift_vfkit_4.16.7_arm64_232.crcbundle
WARN Using custom bundle crc_microshift_vfkit_4.16.7_arm64_232.crcbundle
Successfully configured bundle to ./crc_microshift_vfkit_4.16.7_arm64_232.crcbundle

% crc setup --log-level debug
DEBU CRC version: 2.41.0+1921db
DEBU OpenShift version: 4.16.7
DEBU MicroShift version: 4.16.7
DEBU Running 'crc setup'
DEBU Got bundle path: ./crc_microshift_vfkit_4.16.7_arm64_232.crcbundle
DEBU Failed to parse url: parse "./crc_microshift_vfkit_4.16.7_arm64_232.crcbundle": invalid URI for request
WARN Using custom bundle crc_microshift_vfkit_4.16.7_arm64_232.crcbundle
bundle filename is in unrecognized format
```

Reply:
Hmm, good to know. So this is going to be a little more difficult. What is the meaning of the last field (ex. `2345`)? Or how should I name it in the `FilenameInfo` struct?

Reply:
the last field is only present in a user generated custom bundle, it is just to differentiate that it is not the usual bundle we ship, so maybe we can name it as `customBundleSuffix`?

Reply:
Ok, I used that name.